### PR TITLE
Python: Renable pylint

### PR DESCRIPTION
**Pull request by Fokko**
For some reason, pylint fell through the cracks when migrating from tox to pre-commit

### CODE REVIEW

**Comment by Fokko** [comment]
I've removed the spellcheck because it relies on a C lib enchant that doesn't play very well with virtual-env. It can't find the binding anymore. I spend quite some time to get it to work, but no luck. Also looking at https://github.com/apache/iceberg/pull/4818 I've removed it for now since it is also not very friendly for new people to ask them to install system packages.

**Review by samredai — COMMENTED** [review]
Needs to be rebased but otherwise LGTM

File: python/tests/io/test_pyarrow.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+# pylint: disable=W0212,W0613
```
Comment (dramaticlly) [inline python/tests/io/test_pyarrow.py:17]: Nit: curious on how pylint update its code to warning mapping, from understanding perspective, it's easier to read `disable=protected-access`

**Review by dramaticlly — APPROVED** [review]
LGTM, sad to see dictionary going away

File: python/tests/io/test_pyarrow.py (hunk 1 above)
Comment (Fokko) [reply python/tests/io/test_pyarrow.py]: Thanks, I like that a lot. Just updated the code.

**Review by danielcweeks — APPROVED** [review]

**Pull request merged**